consensus/istanbul/backend/backend_test.go

```diff
@@ -695,41 +680,15 @@ func getGovernance(dbm database.DBManager) *governance.Governance {
 }
 
 func Benchmark_getTargetReceivers(b *testing.B) {
+	_, backend := newBlockChain(1)
+	defer backend.Stop()
+	backend.currentView.Store(&istanbul.View{Sequence: istanbul.TestBlockBeforeHF, Round: big.NewInt(0)})
+
 	// Create ValidatorSet
 	council := getTestCouncil()
 	rewards := getTestRewards()
-
-	// get testing node's address
-	key, _ := crypto.HexToECDSA(PRIVKEY) // This key is to be provided to create backend
-	dbm := database.NewDBManager(&database.DBConfig{DBType: database.MemoryDB})
 	valSet := validator.NewWeightedCouncil(council, nil, rewards, getTestVotingPowers(len(council)), nil, istanbul.WeightedRandom, 21, 0, 0, nil)
-
-	recents, _ := lru.NewARC(inmemorySnapshots)
-	recentMessages, _ := lru.NewARC(inmemoryPeers)
-	knownMessages, _ := lru.NewARC(inmemoryMessages)
-	gov := getGovernance(dbm)
-	backend := &backend{
-		config:            istanbul.DefaultConfig,
-		istanbulEventMux:  new(event.TypeMux),
-		privateKey:        key,
-		address:           crypto.PubkeyToAddress(key.PublicKey),
-		logger:            logger.NewWith(),
-		db:                dbm,
-		commitCh:          make(chan *types.Result, 1),
-		recents:           recents,
-		candidates:        make(map[common.Address]bool),
-		coreStarted:       false,
-		recentMessages:    recentMessages,
-		knownMessages:     knownMessages,
-		rewardbase:        rewards[0],
-		governance:        gov,
-		nodetype:          common.CONSENSUSNODE,
-		rewardDistributor: reward.NewRewardDistributor(gov),
-	}
-	backend.core = istanbulCore.New(backend, backend.config)
-
-	backend.currentView.Store(&istanbul.View{Sequence: big.NewInt(1), Round: big.NewInt(0)})
-	valSet.SetBlockNum(uint64(1))
+	valSet.SetBlockNum(istanbul.TestBlockBeforeHF.Uint64())
```

Comment:
Is it okay to remove those code fragments? Looks many lines are removed.
Does this test work for its purpose?

Reply:
I think it's okay to remove those codes because most lines are same with new function of `consensus/istanbul/backend/backend.go`. Also, i have tested and debugged a lot. Don't worry about this :)